nixos/modules/virtualisation/xen-dom0.nix

```diff
@@ -7,148 +7,290 @@
   ...
 }:
 
-with lib;
-
 let
   cfg = config.virtualisation.xen;
+
+  xenBootBuilderPath = "${
+    pkgs.writeShellApplication {
+      name = "xenBootBuilder";
+      runtimeInputs =
+        (with pkgs; [
+          binutils
+          coreutils
+          findutils
+          gawk
+          gnugrep
+          gnused
+          jq
+        ])
+        ++ lib.lists.optionals (cfg.efi.bootBuilderFlag == "--report") (
+          with pkgs;
+          [
+            bat
+            diffutils
+          ]
+        );
+
+      runtimeEnv = {
+        efiMountPoint = config.boot.loader.efi.efiSysMountPoint;
+      };
+      text = builtins.readFile ./xen-boot-builder.sh;
+    }
+  }/bin/xenBootBuilder";
 in
 
 {
-  imports = [
-    (mkRemovedOptionModule [
-      "virtualisation"
-      "xen"
-      "qemu"
-    ] "You don't need this option anymore, it will work without it.")
+  imports = with lib.modules; [
+    (mkRemovedOptionModule
+      [
+        "virtualisation"
+        "xen"
+        "bridge"
+        "name"
+      ]
+      "The Xen Network Bridge options are currently unavailable. Please set up your own bridge manually."
+    )
+    (mkRemovedOptionModule
+      [
+        "virtualisation"
+        "xen"
+        "bridge"
+        "address"
+      ]
+      "The Xen Network Bridge options are currently unavailable. Please set up your own bridge manually."
+    )
+    (mkRemovedOptionModule
+      [
+        "virtualisation"
+        "xen"
+        "bridge"
+        "prefixLength"
+      ]
+      "The Xen Network Bridge options are currently unavailable. Please set up your own bridge manually."
+    )
+    (mkRemovedOptionModule
+      [
+        "virtualisation"
+        "xen"
+        "bridge"
+        "forwardDns"
+      ]
+      "The Xen Network Bridge options are currently unavailable. Please set up your own bridge manually."
+    )
     (mkRenamedOptionModule
       [
         "virtualisation"
         "xen"
         "qemu-package"
       ]
+      [
+        "virtualisation"
+        "xen"
+        "qemu"
+        "package"
+      ]
+    )
+    (mkRenamedOptionModule
       [
         "virtualisation"
         "xen"
         "package-qemu"
       ]
+      [
+        "virtualisation"
+        "xen"
+        "qemu"
+        "package"
+      ]
+    )
+    (mkRenamedOptionModule
+      [
+        "virtualisation"
+        "xen"
+        "stored"
+      ]
+      [
+        "virtualisation"
+        "xen"
+        "store"
+        "path"
+      ]
+    )
+    (mkRenamedOptionModule
+      [
+        "virtualisation"
+        "xen"
+        "storePath"
```

Comment:
Is `stored` and `storePath` the same option?

Reply:
Actually I have no idea where `storePath` came from. It was probably an early name I gave to `store.path` that was since removed. I've deleted that rename.

And yes, `stored` has been renamed to `store.path` to accommodate the other store-related options.

Reply:
Maybe it will be better named store.package?

Reply:
I don't think so. We're selecting a path, not a package.

Reply:
Oh, I see, this is a path to a binary inside of package.
Maybe xen derivation be better split, but lets keep it this way then.

Reply:
Considering Xen is a monorepo, I believe that would be a lot of work!